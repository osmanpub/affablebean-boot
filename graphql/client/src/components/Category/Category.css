.categoryBox {
  height: 176px;
  width: 212px;
  margin: 21px 14px 6px;
  float: inherit;
}

.categoryImage {
  padding: 1px;
  border: solid 1px #555;
}

.categoryLabel {
  position: absolute;
  background-color: #fff;
  opacity: 0.7;
  height: 40px;
  width: 210px;
  margin: 2px;
}

.categoryLabelText {
  position: absolute;
  line-height: 150%;
  font-size: x-large;
  margin: 3px 10px;
}
