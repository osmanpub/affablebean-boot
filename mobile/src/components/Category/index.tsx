import React from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import {getCategoryIcon, getId} from '../../helpers/utils';
import {Category as CategoryState} from '../../interfaces/categories';

type Props = {
  category: CategoryState;
  setCategoryProduct: Function;
};

export default function Category(props: Props) {
  const {category, setCategoryProduct} = props;
  const {name} = category;

  return (
    <View style={styles.category}>
      <Text>{name}</Text>
      <TouchableWithoutFeedback
        onPress={() => setCategoryProduct(getId(category))}>
        <Image source={getCategoryIcon(name)} />
      </TouchableWithoutFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  category: {
    alignItems: 'center',
    paddingBottom: 24,
  },
});
