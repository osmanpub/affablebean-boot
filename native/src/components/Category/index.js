import React from "react";
import PropTypes from "prop-types";
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";

export default function Category(props) {
  const { category, setCategoryProduct } = props;
  const { id, name } = category;

  const styles = StyleSheet.create({
    category: {
      alignItems: "center"
    },
    categoryImage: {
      padding: 1
    }
  });

  var icon = "";

  switch (name) {
    case "bakery":
      icon = require("./categories/bakery.jpg");
      break;

    case "cereals":
      icon = require("./categories/cereals.jpg");
      break;

    case "dairy":
      icon = require("./categories/dairy.jpg");
      break;

    case "drinks":
      icon = require("./categories/drinks.jpg");
      break;

    case "fruit & veg":
      icon = require("./categories/fruitveg.jpg");
      break;

    case "meats":
      icon = require("./categories/meats.jpg");
      break;

    default:
  }

  function selectCategory() {
    setCategoryProduct(id);
  }

  return (
    <View style={styles.category}>
      <Text>{name}</Text>
      <TouchableOpacity onPress={selectCategory}>
        <Image source={icon} style={styles.categoryImage} />
      </TouchableOpacity>
      <Text />
      <Text />
    </View>
  );
}

Category.propTypes = {
  category: PropTypes.object.isRequired
};
