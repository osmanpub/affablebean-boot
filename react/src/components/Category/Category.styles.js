import styled from "styled-components";

export const CategoriesGreeting = styled.p`
  font-size larger;
`;

export const CategoriesLeft = styled.div`
  text-align: left;
  height: 400px;
  width: 350px;
  float: left;
`;

export const CategoriesRight = styled.div`
  text-align: left;
  height: 400px;
  width: 720px;
  float: left;
`;

export const CategoriesWelcome = styled.div`
  margin: 30px 5px 0 30px;
  line-height: 1.4em;
`;

/* <script type="text/javascript">
$(document).ready(function() {
  $('a.categoryButton').hover(function() {
    $(this).animate({
      backgroundColor : '#b2d2d2'
    });
  }, function() {
    $(this).animate({
      backgroundColor : '#d3ede8'
    });
  });

  $('div.categoryBox').hover(function() {
    var span = this.getElementsByTagName('span');
    $(span[0]).animate({
      opacity : 0.3
    });
    $(span[1]).animate({
      color : 'white'
    });
  }, function() {
    var span = this.getElementsByTagName('span');
    $(span[0]).animate({
      opacity : 0.7
    });
    $(span[1]).animate({
      color : '#444'
    });
  });
});
</script> */
