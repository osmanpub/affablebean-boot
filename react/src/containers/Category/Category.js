import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchCategoryIfNeeded } from "../../rest/category";
import Footer from "../../components/Footer";
import Header from "../../components/Header";

export class Category extends Component {
  componentDidMount() {
    const { dispatch, match } = this.props;
    const { params } = match;
    dispatch(fetchCategoryIfNeeded(params.id));
  }

  // 3 renders on start is too much - investigate
  render() {
    const { categories, match } = this.props;

    return (
      // <Header />
      <Footer />
    );
  }
}

const mapStateToProps = state => {
  const { categories } = state;

  return {
    categories
  };
};

export default connect(mapStateToProps)(Category);
