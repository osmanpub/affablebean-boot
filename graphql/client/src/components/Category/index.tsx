import React from "react";
import { Link } from "react-router-dom";
import { getId } from "../../helpers/utils";
import { Category as CategoryState } from "../../interfaces/categories";
import "./Category.css";

type Props = {
  category: CategoryState;
};

export default function Category(props: Props) {
  const { category } = props;
  const { name } = category;
  const id = getId(category);

  return (
    <span className="categoryBox" data-testid="categoryBox">
      <Link to={`/category/${id}`}>
        <span className="categoryLabel" />
        <span className="categoryLabelText">{name}</span>
        <img
          alt={name}
          className="categoryImage"
          data-cy={`category-${name}`}
          data-testid={`category-${name}`}
          src={`/static/img/categories/${name}.jpg`}
        />
      </Link>
    </span>
  );
}
