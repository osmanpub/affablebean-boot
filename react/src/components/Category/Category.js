import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "./Category.css";

export function Category(props) {
  const { category } = props;
  const { name } = category;

  return (
    <span className="categoryBox">
      <Link to={`/category/${category.id}`}>
        <span className="categoryLabel" />
        <span className="categoryLabelText">{name}</span>
        <img
          src={`/static/img/categories/${name}.jpg`}
          alt={name}
          className="categoryImage"
        />
      </Link>
    </span>
  );
}

Category.propTypes = {
  category: PropTypes.object.isRequired
};
