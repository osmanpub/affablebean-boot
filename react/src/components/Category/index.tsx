import React from "react";
import { Link } from "react-router-dom";
import "./Category.css";
import { Category as CategoryState } from "../../interfaces/categories";

type Props = {
  category: CategoryState;
};

export default function Category(props: Props) {
  const { category } = props;
  const { name } = category;

  return (
    <span className="categoryBox">
      <Link to={`/category/${category.id}`}>
        <span className="categoryLabel" />
        <span className="categoryLabelText">{name}</span>
        <img
          data-cy={`category-${name}`}
          src={`/static/img/categories/${name}.jpg`}
          alt={name}
          className="categoryImage"
        />
      </Link>
    </span>
  );
}
