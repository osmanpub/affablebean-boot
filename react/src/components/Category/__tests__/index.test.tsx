import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Category from "..";
import { shallow } from "enzyme";

describe("<Category />", () => {
  const category = {
    id: 1,
    name: "dairy"
  };

  it("renders correctly", () => {
    const component = renderer
      .create(
        <Router>
          <Category category={category} />
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("shallow testing", () => {
    const component = shallow(<Category category={category} />);
    // console.warn(component.props)
    // expect(component.find(".categoryImage")).to.have.lengthOf(1);
  });
});
