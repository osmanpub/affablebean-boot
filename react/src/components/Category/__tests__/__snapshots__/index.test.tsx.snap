// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> renders correctly 1`] = `
<span
  className="categoryBox"
  data-testid="categoryBox"
>
  <a
    href="/category/1"
    onClick={[Function]}
  >
    <span
      className="categoryLabel"
    />
    <span
      className="categoryLabelText"
    >
      dairy
    </span>
    <img
      alt="dairy"
      className="categoryImage"
      data-cy="category-dairy"
      src="/static/img/categories/dairy.jpg"
    />
  </a>
</span>
`;
